lib/scala/logging-service-logback/src/main/java/org/enso/logger/LogbackSetup.java

```diff
@@ -101,12 +105,12 @@ public boolean setupSocketAppender(
         Level targetLogLevel;
         // Modify log level if we were asked to always log to a file.
         // The receiver needs to get all logs (up to `trace`) so as to be able to log all verbose messages.
-        if (config.logToFile()) {
+        if (logToFileEnabled()) {
             targetLogLevel = Level.TRACE;
```

Comment:
Shouldn't we use the configured log level instead?

```suggestion
            targetLogLevel = SOME_MAXIMUM(config.logToFile().logLevel(), logLevel);
```

---

I'm worried that processing the TRACE level logs may already incur a cost. Even if we 'just' process them, we may be sending them thru a socket to a parent process. That is likely cheaper than writing to disk, but is still not 0-cost.

If file log level is just `debug` and console is `info`, then we should not be processing the `trace` level warnings _at all_.

Reply:
I assume we have the same configuration at receiver and sender. I guess this may not always be the case.

But I would risk assuming the config _is_ the same - any config mismatch will be responsibility of the user (usually developer) changing the configurations. The regular users will be relying on the defaults anyway. And changing the env-var should also affect both receiver and sender in most typical configurations.

Reply:
I'd create a config section for the socket appended. This way there will be an opportunity to override the log level with the env variable

Reply:
I like the maximal level inference. I think it would suffice for our use-case.